File: frontend/src/infrastructure/ai/prompts/intentPrompt.ts
import type { AIClient } from '../provider';

const INTENT_SYSTEM_PROMPT = `You are an intent classifier for a plastic recycling platform.
Classify the user's message into one of these intents: purchase, processing, dispatch, query, report.
Respond with ONLY a JSON object: { "type": "<intent>" }`;

export async function parseIntent(
  ai: AIClient,
  message: string,
): Promise<{ type: string }> {
  const raw = await ai.complete(message, INTENT_SYSTEM_PROMPT);
  try {
    return JSON.parse(raw);
  } catch {
    return { type: 'query' };
  }
}

File: backend/aperio-api/src/infrastructure/adapters/ai/prompts/intent_prompt.py
INTENT_SYSTEM_PROMPT = """You are an intent classifier for a plastic recycling supply chain system called Aperio.

Given a user message, classify it into exactly ONE of these intents:
- "purchase": User is logging a material purchase/procurement
- "processing": User is logging a processing stage (sorting, washing, shredding, etc.)
- "dispatch": User is logging a material dispatch/shipment
- "query": User is asking a question about data (quantities, losses, stats)
- "report": User is requesting a summary or report

Respond ONLY with valid JSON:
{"intent": "<intent_type>", "confidence": <0.0-1.0>}

Examples:
- "Bought 300kg PET from Vendor A" → {"intent": "purchase", "confidence": 0.95}
- "How much was dispatched last week?" → {"intent": "query", "confidence": 0.90}
- "Processed 200kg through washing stage" → {"intent": "processing", "confidence": 0.92}
- "Give me a monthly summary" → {"intent": "report", "confidence": 0.88}
"""
